src/plugins/collapsibleColumns/collapsibleColumns.js

```diff
@@ -408,53 +440,131 @@ class CollapsibleColumns extends BasePlugin {
-    const childHeaders = this.nestedHeadersPlugin.getChildHeaders(coords.row, coords.col);
-    let nextLevel = level + 1;
-    let childColspanLevel = colspanArray[nextLevel];
-    let firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
-
-    while (firstChildColspan === currentHeaderColspan && nextLevel < this.columnHeaderLevelCount) {
-      nextLevel += 1;
-      childColspanLevel = colspanArray[nextLevel];
-      firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
-    }
+    let destinationCollapsedColumns = currentCollapsedColumns;
+    let collapsePossible;
 
-    rangeEach(firstChildColspan, currentHeaderColspan - 1, (i) => {
-      const colToHide = coords.col + i;
+    arrayEach(coords, (currentCoords) => {
+      if (currentCoords.row) {
+        currentCoords.row = parseInt(currentCoords.row, 10);
+      }
+      if (currentCoords.col) {
+        currentCoords.col = parseInt(currentCoords.col, 10);
+      }
 
-      switch (action) {
-        case 'collapse':
-          if (!this.hiddenColumnsPlugin.isHidden(colToHide)) {
-            hiddenColumns.push(colToHide);
-          }
+      if (!this.collapsableCoordsList[currentCoords.row] ||
+        (this.collapsableCoordsList[currentCoords.row] && !this.collapsableCoordsList[currentCoords.row][currentCoords.col])) {
+        collapsePossible = false;
 
-          break;
-        case 'expand':
-          if (this.hiddenColumnsPlugin.isHidden(colToHide)) {
-            hiddenColumns.splice(hiddenColumns.indexOf(colToHide), 1);
-          }
+        return false;
+      }
+
+      collapsePossible = this.collapsableCoordsList[currentCoords.row][currentCoords.col];
+
+      const colspanArray = this.nestedHeadersPlugin.colspanArray;
+      const level = this.nestedHeadersPlugin.rowCoordsToLevel(currentCoords.row);
+      const currentHeaderColspan = colspanArray[level][currentCoords.col].colspan;
+      const childHeaders = this.nestedHeadersPlugin.getChildHeaders(currentCoords.row, currentCoords.col);
+      let nextLevel = level + 1;
+      let childColspanLevel = colspanArray[nextLevel];
+      let firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
 
-          break;
-        default:
-          break;
+      while (firstChildColspan === currentHeaderColspan && nextLevel < this.columnHeaderLevelCount) {
+        nextLevel += 1;
+        childColspanLevel = colspanArray[nextLevel];
+        firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
       }
+
+      rangeEach(firstChildColspan, currentHeaderColspan - 1, (i) => {
+        const colToHide = currentCoords.col + i;
+
+        switch (action) {
+          case 'collapse':
+            if (!this.hiddenColumnsPlugin.isHidden(colToHide)) {
+              hiddenColumns.push(colToHide);
+            }
+
+            destinationCollapsedColumns = Array.from(hiddenColumns);
```

Comment:
When I have already collapsed for instance 3 columns and try to collapse 4th one (for different layer) it seems that this `destinationCollapsedColumns` variable is created four times but only last assignment is passed to the hook? If this is the case then it could be improved.